lib/beaker/subcommand.rb

```diff
@@ -49,55 +53,134 @@ def initialize(*args)
     class_option :'exclude-tags', :type => :string, :group => 'Beaker run'
     class_option :'xml-time-order', :type => :boolean, :group => 'Beaker run'
 
-    desc "init HYPERVISOR", "Initialises the beaker test environment configuration"
+    desc "init BEAKER_RUN_OPTIONS", "Initializes the required configuration for Beaker subcommand execution"
     option :help, :type => :boolean, :hide => true
     long_desc <<-LONGDESC
+      Initializes the required .beaker configuration folder. This folder contains
+      a subcommand_options.yaml file that is user-facing; altering this file will
+      alter the options subcommand execution.
 
-      Initialises a test environment configuration for a specific hypervisor
-
-      Supported hypervisors are: vagrant (default), vmpooler
-
+      Also modified in this operation is the .subcommand_state file that is used
+      by beaker to determine subcommand state. You as a user should never have to
+      edit this file.
     LONGDESC
```

Comment:
@ferglor I agree, I don't like how it's telling the user what's going on under the hood either & we shouldn't tell them about the `.subcommand_state` file. Also, I think we should tell them about supported hypervisors, since those are needed in the 'new user' use case, which is when we want to be most verbose.

To your point about the file names @ferglor, I disagree. The folder these files live in is `.beaker`, & these files are subcommand-specific, so they should be named as such. If we leave them as general as `beaker`, then people will believe they should be usable outside of the subcommand workflow. "Subcommand" isn't implementation-specific either, it's the most general name for what we're doing & we could implement subcommands in any number of ways.